src/ScottPlot5/ScottPlot/DataSource/SignalList.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace ScottPlot.DataSource;
+
+/// <summary>
+/// This data source has evenly-spaced Xs and Ys stored in a fixed-length array.
+/// Changes made to the contents of the array will appear when the plot is rendered.
+/// </summary>
+public class SignalList : ISignalSource
+{
+    public double Period { get; set; }
+    public double XOffset { get; set; }
+    public List<double> Ys { get; }
+
+    public int Count => Ys.Count;
+
+    double IReadOnlyList<double>.this[int index] => Ys[index];
+
+    public SignalList(List<double> ys, double period)
```

Comment:
I think this should be `IList<double> ys`. The current signature disallows arrays.

Reply:
Oh I see, we have a `SignalArrary` and a `SignalList`. I'm not sure why? Is the only difference the type of the backing collection? If there is a larger reason why they're separate, it may still be worth generalizing the code into something like this:

```cs
public abstract class SignalCollection<T> {}
public class SignalList : SignalCollection<List<double>> {}
public class SignalArray : SignalCollection<double[]> {}
```

Reply:
> we have a SignalArrary and a SignalList. I'm not sure why?

This is my line of thinking...

* Users who want to do special stuff instantiate a concrete class of their choice and pass it into a plottable
  * If users want to plot an existing array in memory without copying, they can
  * If users want to plot an existing list they can add to, they can

* Plottables only access data via the interface. The interface describes data as read only.